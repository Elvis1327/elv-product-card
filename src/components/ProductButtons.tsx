import React from 'react';
import { CSSProperties, useCallback, useContext } from "react";

import { ProductContext } from "./ProductCard";
import styles from '../styles/styles.module.css';

interface Props {
    className?: string;
    optionColor?: string;
    style?: CSSProperties;
}

export const ProductButtons = ({className, optionColor, style}: Props) => {
    const { counter, increaseBy, maxCount } = useContext(ProductContext); 
    
    const isMaxReached = useCallback(() => !!maxCount && counter === maxCount,
        [counter, maxCount],
    )


    return (
        <div className={styles.buttonsContainer} style={style}>
            <button
                className={`${styles.buttonMinus} ${optionColor}`}
                onClick={() => increaseBy(-1)}
            >
                    -
            </button>
            <div
                className={styles.countLabel}
            >
                {counter}
            </div>
            <button
                className={`${styles.buttonAdd} ${ isMaxReached() && styles.disabled} ${className}`}
                onClick={() => increaseBy(1)}
            >
                +
            </button>
        </div>
    )
};