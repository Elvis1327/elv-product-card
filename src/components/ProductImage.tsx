import React from 'react';
import { useContext, CSSProperties } from "react";

import noImage from '../assets/no-image.jpg';
import { ProductContext } from "./ProductCard";
import styles from '../styles/styles.module.css';

interface Props {
    img?:string;
    className?: string;
    style?: CSSProperties;
}

export const ProductImage = ({img, className, style}: Props) => {
    const { product } = useContext(ProductContext);

    let imgToShow: string = '';
    if(img) {
        imgToShow = img;
    }else if(product.img){
        imgToShow = product.img
    }else{
        imgToShow = noImage
    }

    return (
        <img 
                className={`${styles.productImg} ${className}`}
                src={imgToShow} 
                alt="img-pic"
                style={style}
            />
    )
};