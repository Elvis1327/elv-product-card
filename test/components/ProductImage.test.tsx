import React from 'react';
import renderer from 'react-test-renderer';

import { ProductImage } from '../../src/components/index';


describe('ProductImage', () => {
    test('Debe de mostrar el componente correctamente con la Imagen del producto', () => {
        const wrapper = renderer.create(
            <ProductImage img="https://res.cloudinary.com/dtp3gqean/image/upload/v1623042649/probar_qjetgg.png" />
        );
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    
})
