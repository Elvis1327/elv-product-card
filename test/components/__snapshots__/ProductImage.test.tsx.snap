// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Debe de mostrar el componente correctamente con la Imagen del producto 1`] = `
<img
  alt="img-pic"
  className="productImg undefined"
  src="https://res.cloudinary.com/dtp3gqean/image/upload/v1623042649/probar_qjetgg.png"
/>
`;
